intermediate-goals.Rmd

```diff
@@ -0,0 +1,148 @@
+# Intermediate Outline {#intermediate-outline}
+
+This outline describes the questions that the intermediate course will answer.
+It assumes learners have completed the novice course, 
+which means they know:
+
+-   how to write code (in Python or R) that reads, analyses and plots a dataset,
+-   how to write and run unit tests,
+-   how to publish a static website using GitHub Pages,
+-   how to organize the files in a small project,
+-   the basics of licensing, and
+-   the basics of good behavior.
+
+## The Unix Shell
+
+Based on <http://swcarpentry.github.io/shell-novice/>
+and <https://merely-useful.github.io/still-magic/en/shell.html>
+
+-   How do I navigate my filesystem?
+    -   moving about (cd, ls, ., ..)
+    -   files and directories (ls, cp, mkdir, mv, rm)
+    -   paths (absolute, relative, special) 
+-   How do I examine and/or edit a text file?
+    -   difference between text and binary files
+    -   running a text editor
+    -   operating on text (cat, wc, head, tail, etc)
+    -   finding things (find, grep)
+-   How do I combine multiple commands?
+    -   pipes and filters
+    -   loops
+-   Variables
+    -   define a variable
+    -   use a variable
+    -   what's with all the quotes?
+-   Shell scripts
+    -   read/write/execute file permissions
+    -   `#!` lines
+    -   using command-line arguments in scripts
+-   Shortcuts
+    -   tab completion
+    -   previous commands (up arrow, control-r)
+    -   history
+-   Getting help
+    -   --help, man
+-   Working remotely
+    -   scp and ssh
+    -   key pairs
+
+## Automating analyses
+
+Based on <http://swcarpentry.github.io/make-novice/>
+and <https://merely-useful.github.io/still-magic/en/automate.html>
+
+-   How do I write a simple Makefile? 
+-   How can I abbreviate the rules in my Makefiles? 
+-   How can I write a Makefile to update things when my scripts have changed rather than my input files?
+-   How can I define rules to operate on similar files? 
+    -   Pattern rules
+-   How can I eliminate redundancy in my Makefiles?
+    -   Variables
+    -   Functions
+-   How should I document a Makefile?
+-   What other tools can I use?
+    -   drake for R
+    -   snakemake for Python
+
+## Reusable software
```

Comment:
Not entirely clear what "Reusable software" section is about. Would it be like defensive programming and including "stop" functions in the code? How to create packages? I can see that this section has a definite feel from the Python world, but I'm struggling to make a comparison to the R world...